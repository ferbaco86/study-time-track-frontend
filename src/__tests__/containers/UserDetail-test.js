import {
  render, screen, act, waitFor,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import UserDetail from '../../containers/UserDetail';

const store = createStore(() => ({
  user: {
    user: {
      id: 1,
      username: 'ferbaco',
      sessions: [
        {
          id: 1,
          user_id: 1,
          title: 'Pelusa',
          created_at: '2021-04-01T00:50:52.332Z',
          total_time: null,
        },
        {
          id: 2,
          user_id: 1,
          title: 'React',
          created_at: '2021-04-01T00:53:27.290Z',
          total_time: null,
        },
        {
          id: 4,
          user_id: 1,
          title: 'C#',
          created_at: '2021-04-01T00:58:29.552Z',
          total_time: null,
        },
      ],
    },
  },
}),
applyMiddleware(thunk));

describe('Rendering component', () => {
  it('creates a UserDetail component', async () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <UserDetail />
          </Provider>
        </BrowserRouter>,
      );
    });
    await waitFor(() => {
      expect(screen.getByText('ferbaco'));
    });
  });
});
