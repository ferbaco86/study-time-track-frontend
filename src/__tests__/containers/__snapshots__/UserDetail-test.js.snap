// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail renders correctly 1`] = `
<div>
  <h1
    className="sc-eCssSg gJiiYa"
  >
    ferbaco
  </h1>
  <a
    className="sc-pFZIQ hRelQs"
    href="/sessionDetail/1"
    onClick={[Function]}
  >
    <div
      className="sc-jSgupP gdcAjd"
    >
      <h4
        className="sc-gKsewC ghVxdN"
      >
        3/31/2021
      </h4>
      <div
        className="sc-fubCfw kdoYVX"
      >
        <h2
          className="sc-iBPRYJ epREFZ"
        >
          Pelusa
        </h2>
      </div>
    </div>
  </a>
  <a
    className="sc-pFZIQ hRelQs"
    href="/sessionDetail/2"
    onClick={[Function]}
  >
    <div
      className="sc-jSgupP gdcAjd"
    >
      <h4
        className="sc-gKsewC ghVxdN"
      >
        3/31/2021
      </h4>
      <div
        className="sc-fubCfw kdoYVX"
      >
        <h2
          className="sc-iBPRYJ epREFZ"
        >
          React
        </h2>
      </div>
    </div>
  </a>
  <a
    className="sc-pFZIQ hRelQs"
    href="/sessionDetail/4"
    onClick={[Function]}
  >
    <div
      className="sc-jSgupP gdcAjd"
    >
      <h4
        className="sc-gKsewC ghVxdN"
      >
        3/31/2021
      </h4>
      <div
        className="sc-fubCfw kdoYVX"
      >
        <h2
          className="sc-iBPRYJ epREFZ"
        >
          C#
        </h2>
      </div>
    </div>
  </a>
</div>
`;
